.github/workflows/DetectCDKBootstrapVersionChanges.yml

```diff
@@ -1,49 +1,107 @@
-    - name: Get Staging Bucket Deletion Policy
-      id: stagingBucketDeletionPolicy
+
+    - name: Create temporary directory
+      run: mkdir -p temp_cdk
+
+    - name: Save New CDK Bootstrap Template
+      working-directory: temp_cdk
       run: |
-        echo "deletion-policy=$(yq '.Resources.StagingBucket.DeletionPolicy' 'src/AWS.Deploy.Orchestration/CDK/CDKBootstrapTemplate.yaml')" >> $GITHUB_OUTPUT
-    - name: Fail If Update/Replace Policy Not 'Delete'
-      if: steps.stagingBucketUpdateReplacePolicy.outputs.update-replace-policy != 'Delete'
+        cdk acknowledge 32775
+        cdk bootstrap --show-template > newTemplate.yml
+
+    - name: Update Template with Required Policies
+      working-directory: temp_cdk
       run: |
-        echo "The 'UpdateReplacePolicy' of the 'StaginBucket' in the CDK bootstrap template should be 'Delete'."
-        exit 1
-    - name: Fail If Deletion Policy Not 'Delete'
-      if: steps.stagingBucketDeletionPolicy.outputs.deletion-policy != 'Delete'
+        yq eval '.Resources.StagingBucket.UpdateReplacePolicy = "Delete"' -i newTemplate.yml
+        yq eval '.Resources.StagingBucket.DeletionPolicy = "Delete"' -i newTemplate.yml
+
+    - name: Check for version changes
+      id: check_version
       run: |
-        echo "The 'DeletionPolicy' of the 'StaginBucket' in the CDK bootstrap template should be 'Delete'."
-        exit 1
-    - name: Save New CDK Bootstrap Template
+        OLD_VERSION=$(yq eval '.Resources.CdkBootstrapVersion.Properties.Value' src/AWS.Deploy.Orchestration/CDK/CDKBootstrapTemplate.yaml)
+        NEW_VERSION=$(yq eval '.Resources.CdkBootstrapVersion.Properties.Value' temp_cdk/newTemplate.yml)
+        
+        if [ "$OLD_VERSION" != "$NEW_VERSION" ]; then
+          echo "Version changed from $OLD_VERSION to $NEW_VERSION"
+          echo "version_changed=true" >> $GITHUB_OUTPUT
+          echo "new_version=$NEW_VERSION" >> $GITHUB_OUTPUT
+        else
+          echo "No version change detected"
+          echo "version_changed=false" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Update CDK Bootstrap Template
+      if: steps.check_version.outputs.version_changed == 'true'
       run: |
-        cdk bootstrap --show-template > newTemplate.yml
-    - name: Get Latest CDK Bootstrap Version
-      id: latestBootstrapVersion
+        cp temp_cdk/newTemplate.yml src/AWS.Deploy.Orchestration/CDK/CDKBootstrapTemplate.yaml
+
+    - name: Generate change file
+      if: steps.check_version.outputs.version_changed == 'true'
+      env:
+        NEW_VERSION: ${{ steps.check_version.outputs.new_version }}
       run: |
-        echo "latest-version=$(yq '.Resources.CdkBootstrapVersion.Properties.Value' 'newTemplate.yml')" >> $GITHUB_OUTPUT
-    - name: Get Current CDK Bootstrap Version
-      id: currentBootstrapVersion
+        dotnet tool install -g autover --version 0.0.22
```

Comment:
use 0.0.24

Reply:
ill update it - i copied the same one from sync-to-dev-to-main workflow